internal/repository/advancer.go

```diff
@@ -0,0 +1,191 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cartesi/rollups-node/internal/node/machine/nodemachine"
+	. "github.com/cartesi/rollups-node/internal/node/model"
+	"github.com/jackc/pgx/v5"
+)
+
+var ErrAdvancerRepository = errors.New("advancer repository error")
+
+type AdvancerRepository struct{ *Database }
+
+func (repository *AdvancerRepository) GetInputs(
+	ctx context.Context,
+	appAddresses []Address,
+) (map[Address][]*Input, error) {
+	result := map[Address][]*Input{}
+	if len(appAddresses) == 0 {
+		return result, nil
+	}
+
+	query := fmt.Sprintf(`
+	SELECT   id, application_address, index, raw_data, block_number, status
+	FROM     input
+    WHERE    status = 'NONE'
+    AND      application_address IN (%s) 
+    ORDER BY index ASC, application_address
```

Comment:
The advance runner is not using the `index`, `block_number` or `status` anywhere, we shouldn't fetch them here.